test/synthetic/test_synthetic_data.py

```diff
@@ -0,0 +1,75 @@
+import unittest
+
+import numpy as np
+
+from snorkel.labeling.apply import LFApplier
+from snorkel.synthetic.synthetic_data import (
+    generate_mog_dataset,
+    generate_single_feature_LFs,
+)
+
+
+class TestGenerateMOGDataset(unittest.TestCase):
+    def setUp(self) -> None:
+        self.n = 1000
+        self.d = 20
+        self.data = generate_mog_dataset(self.n, self.d)
+
+    def test_output_types(self) -> None:
+        # Test x
+        x = self.data[0].x
+        self.assertIsInstance(x, np.ndarray)
+        self.assertEqual(x.shape[0], self.d)
+
+        # Test y
+        self.assertIsInstance(self.data[0].y, np.integer)
+
+    def test_dataset_means(self) -> None:
```

Comment:
Fine for now, but we probably want to run this for multiple setUp calls since its rand

Reply:
For this simple example, should always work- but down the road sure